test/stdlib/Inputs/FoundationBridge/FoundationBridge.h

```diff
@@ -85,4 +85,9 @@ static inline BOOL NSStringBridgeTestEqual(NSString * _Nonnull a, NSString * _No
   return [a isEqual:b];
 }
 
+static inline NSString *getNSStringWithUnpairedSurrogate() {
+  unichar chars[1] = { 0xD800 };
+  return [NSString stringWithCharacters:chars length:1];
+}
+
```

Comment:
How well will this test hold up when you eagerly bridge smol Unicode strings? Would it silently continue passing but no longer testing the relevant portion?

Reply:
Hmm possibly yes. I could try to use the trivial NSString subclass trick, but I'm wary of doing that when we're trying to test the behavior of the default NSString implementation.

Reply:
This would still repro if the string were longer, right? We could do a longer string ending in a surrogate. BTW, what happens if the unpaired surrogate is in the middle?

Reply:
Ah true, I think that should work. If it's in the middle it still repros